docs/src/charm/howto/etcd.md

```diff
@@ -0,0 +1,132 @@
+# How to integrate Canonical Kubernetes with etcd
+
+Integrating **etcd** with your Canonical Kubernetes deployment provides a
+robust, distributed key-value store that is essential for storing critical
+data needed for Kubernetes' clustering operations. This guide will walk you
+through the process of deploying Canonical Kubernetes with an external etcd
+cluster.
+
+## What you will need
+
+- A Juju controller with access to a cloud environment (see the [Juju setup]
+  guide for more information).
+
+```{warning} Once you deploy your Canonical Kubernetes cluster with a
+particular datastore, you cannot switch to a different datastore
+post-deployment. Planning for your datastore needs ahead of time is
+crucial, particularly if you opt for an external datastore like **etcd**.
+```
+
+## Preparing the Deployment
+
+1. **Creating the Deployment Model**:
+  Begin by creating a Juju model specifically for your Canonical Kubernetes
+  cluster deployment.
+
+  ```bash
+  juju add-model my-cluster
+  ```
+2. **Deploying Certificate Authority**:
+  etcd requires a secure means of communication between its components.
+  Therefore, we require a certificates authority such as [EasyRSA][easyrsa-charm]
+  or [Vault][vault-charm]. Check the respective charm documentation for detailed
+  instructions on how to deploy a certificates authority. In this guide, we will
+  be using EasyRSA.
+
+  ```bash
+  juju deploy easyrsa
+  ```
+
+## Deploying etcd
+
+- **Single Node Deployment**:
+  - To deploy a basic etcd instance on a single node, use the command:
+
+    ```bash
+    juju deploy etcd
+    ```
+
+    This setup is straightforward but not recommended for production environments
+    due to lack of high availability.
+
+- **High Availability Setup**:
+  - For environments where high availability is crucial, deploy etcd across at
+    least three nodes:
+
+    ```bash
+    juju deploy etcd -n 3
+    ```
+
+    This ensures that your etcd cluster remains available even if one node fails.
+
+## Integrating etcd with EasyRSA
+
+Now you have to integrate etcd with your certificate authority; this will issue
+the required certificates for secure communication between etcd and your
+Canonical Kubernetes cluster:
+
+```bash
+juju integrate etcd easyrsa
+```
+
+## Deploying Canonical Kubernetes
+Deploy the control plane units of Canonical Kubernetes with the command:
+
+```bash
+juju deploy k8s --config datastore=etcd -n 3
+```
```

Comment:
i don't know if it's possible, but we should really encourage `juju expose k8s` in our How-Tos.  Without it people will be frustrated with the experience on public clouds